### DISCUSSION TITLE
Question : Bounded delay

### DISCUSSION THREAD

**Discussion by zinebbe** [apache/incubator-mxnet#17664] [Q&A] [answered]
Hello,
I would like to use the parameter server along with mxnet.
Could you please let me know if it is possible to change the bounded delay on the parameter server.
I am aware that the bounded delay is not implemented in the ps-lite so I am trying to use the initial version of paramater server.
Thank in advance for your answer.

**Comment by stu1130** [comment]
[@mxnet-label-bot add question]

**Comment by stu1130** [comment]
@eric-haibin-lin could you take a look at this question?

**Comment by zinebbe** [comment]
@eric-haibin-lin any update please?
Thanks again for your help!

**Comment by eric-haibin-lin** [comment]
I don't think mxnet is integrated with the parameter server library. In ps-lite, you can probably leverage the timestamp in the message and add parameter versioning logic on the server side, but you need to implement it yourself

**Comment by zinebbe** [comment]
Thank you very much for your answer, is it possible to use the mxnet with the parameter server since the bounded delay is implemented there? If so, any tip on how to proceed?
Thanks again

**Comment by eric-haibin-lin** [comment, marked as answer]
you can keep parameter version by implementing a custom updater/optimizer https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/optimizer/optimizer.py#L1361 which is registered on kvstore server
